### PR TITLE
feat: Adding `stylecheck` lint

### PR DESCRIPTION
**Pull request by yhakbar**
## Description

Adds `stylecheck` lints. This ensures that we use idiomatic naming for variables and functions, etc.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added `stylecheck` lints.

### CODE REVIEW

**Comment by yhakbar** [comment]
Moving to in-draft to avoid conflicts with #3338. This does not need to be merged in with any urgency.

File: test/integration_json_test.go [hunk 1]
```diff
@@ -476,9 +476,9 @@ func TestRenderJsonMetadataTerraform(t *testing.T) {
 func TestRenderJsonMetadataDependencyModulePrefix(t *testing.T) {
 	t.Parallel()
 
-	tmpEnvPath := copyEnvironment(t, testFixtureRenderJsonMetadata)
+	tmpEnvPath := copyEnvironment(t, testFixtureRenderJSONMetadata)
```
Comment (denis256) [inline test/integration_json_test.go:479]: Looks like there is an inconsistency in the naming convention for test constants:
testFixtureRenderJSONMetadata - private variable
TestFixtureGetTerragruntSourceHcl - public variable

**Comment by yhakbar** [comment]
Looks like I have more work to do to unwind merge conflicts with `main`.

**Review by levkohimins — APPROVED** [review]
Awesome! You did a lot of hard work.

**Comment by yhakbar** [comment]
Almost ready to be marked ready for review, but will wait until some logging improvements go through to avoid making that merge harder.

**Review by levkohimins — CHANGES_REQUESTED** [review]

**Review by levkohimins — APPROVED** [review]
LGTM!

**Pull request merged**